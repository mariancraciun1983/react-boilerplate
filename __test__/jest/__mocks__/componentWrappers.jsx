import React  from 'react';
import clone from 'clone-deep';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import mockStore  from './store';


const { fullStore } = mockStore;

const genericWrapper = (children) => {
  const mockStore = configureStore();
  const initialState = clone(fullStore);
  const store = mockStore(initialState);
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  )
};

export {
  genericWrapper
};
