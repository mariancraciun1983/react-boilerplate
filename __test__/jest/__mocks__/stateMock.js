export default {
  app: {
    state: "loaded"
  },
  auth: {
    authenticated: true,
    token: "abc123",
    user: {
      name: "John Doe",
      email: "john@example.com",
      avatar: "https://dummyimage.com/100x100.png/5fa2d1/ffffff"
    }
  },
  genres: {
    list: [
      {
        name: "Comedy",
        slug: "comedy"
      },
      {
        name: "Adventure",
        slug: "adventure"
      }
    ]
  },
  movies: {
    list: [
      {
        id: "0b7b8b05-6c88-4517-a969-adf89a46bc0a",
        title: "Lost Palace",
        genre: ["comedy"],
        image: "http://dummyimage.com/100x100.png/cc0000/ffffff",
        description:
          "Donec diam neque, vestibulum eget, vulputate ut, ultrices vel, augue. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec pharetra, magna vestibulum aliquet ultrices, erat tortor sollicitudin mi, sit amet lobortis sapien sapien non mi. Integer ac neque.\n\nDuis bibendum. Morbi non quam nec dui luctus rutrum. Nulla tellus.\n\nIn sagittis dui vel nisl. Duis ac nibh. Fusce lacus purus, aliquet at, feugiat non, pretium quis, lectus.\n\nSuspendisse potenti. In eleifend quam a odio. In hac habitasse platea dictumst.\n\nMaecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat.",
        price: 2.39,
        slug: "gunner-palace",
        title_lower: "lost palace"
      },
      {
        id: "f01187b5-6fe8-465b-a0b8-3764a94ae225",
        title: "Love is Eternal While It Lasts",
        genre: ["comedy", "romance"],
        image: "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
        description:
          "Curabitur gravida nisi at nibh. In hac habitasse platea dictumst. Aliquam augue quam, sollicitudin vitae, consectetuer eget, rutrum at, lorem.\n\nInteger tincidunt ante vel ipsum. Praesent blandit lacinia erat. Vestibulum sed magna at nunc commodo placerat.\n\nPraesent blandit. Nam nulla. Integer pede justo, lacinia eget, tincidunt eget, tempus vel, pede.",
        price: 9.5,
        slug: "love-is-eternal-while-it-lasts",
        title_lower: "love is eternal while it lasts"
      },
      {
        id: "23ad0474-8b8d-420d-a2cc-dd54aa6ed589",
        title: "Lost World, The",
        genre: ["romance"],
        image: "http://dummyimage.com/100x100.png/cc0000/ffffff",
        description:
          "In congue. Etiam justo. Etiam pretium iaculis justo.\n\nIn hac habitasse platea dictumst. Etiam faucibus cursus urna. Ut tellus.\n\nNulla ut erat id mauris vulputate elementum. Nullam varius. Nulla facilisi.\n\nCras non velit nec nisi vulputate nonummy. Maecenas tincidunt lacus at velit. Vivamus vel nulla eget eros elementum pellentesque.\n\nQuisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.",
        price: 7.4,
        slug: "lost-world-the",
        title_lower: "lost world, the"
      }
    ]
  },
  cart: {
    dict: {
      "0b7b8b05-6c88-4517-a969-adf89a46bc0a": 1,
      "23ad0474-8b8d-420d-a2cc-dd54aa6ed589": 2
    },
    saved: false
  },
  filters: {
    string: "Lo",
    string_lower: "lo"
  },
  ui: {
    language: "en",
    version: 2,
    theme: "light"
  },
  routing: {
    locationBeforeTransitions: {
      pathname: "/genre/comedy",
      search: "",
      hash: "",
      key: "sb1up4"
    }
  },
  form: {}
};
