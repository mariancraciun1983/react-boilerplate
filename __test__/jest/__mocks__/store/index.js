import clone from 'clone-deep';

import * as FullData from '../../../../__mock__/api';

const defaultStore = {
  app: {
    state: 'loading',
  },
  auth: {
    authenticated: false,
    token: null,
    user: null,
  },
  cart: {
    dict: {},
    saved: true,
  },
  filters: {
    string: '',
    string_lower: '',
  },
  genres: {
    list: null,
  },
  movies: {
    list: null,
  },
  ui: {
    language: 'en',
    version: 0,
    theme: 'light',
  },
  routing: {},
  form: {},
};

const fullStore = clone(defaultStore);
fullStore.state = 'loaded';
fullStore.genres.list = FullData.genres;
fullStore.movies.list = FullData.movies.slice(0,200);
export default {
  defaultStore,
  fullStore,
}
