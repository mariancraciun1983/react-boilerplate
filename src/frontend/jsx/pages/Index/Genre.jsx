import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import { genreBySlug } from '../../../core/selectors';

import Nav from './components/Nav';
import Movies from './components/Movies';

const stateToProps = (state, props) => {
  // eslint-disable-next-line camelcase
  const { genre_slug } = props.match.params;
  return {
    genre: genreBySlug(state, { genre_slug }),
  };
};
const actionsToProps = (dispatch) => ({
  redirect: (pathname) => dispatch(push(pathname)),
});

class Genre extends React.PureComponent {
  componentDidMount() {
    const { genre } = this.props;
    if (!genre) {
      const { redirect } = this.props;
      redirect('/error/404');
    }
  }

  render() {
    const { genre } = this.props;
    if (!genre) return null;
    return (
      <div className="d-flex">
        <div className="left-nav">
          <Nav genre={genre} />
        </div>
        <div className="flex-grow-1">
          <Movies genre={genre} />
        </div>
      </div>
    );
  }
}

Genre.propTypes = {
  redirect: PropTypes.func.isRequired,
  genre: PropTypes.shape({
    slug: PropTypes.string,
  }),
};
Genre.defaultProps = {
  genre: null,
};

export const Component = Genre;
export default connect(
  stateToProps,
  actionsToProps,
)(Genre);
