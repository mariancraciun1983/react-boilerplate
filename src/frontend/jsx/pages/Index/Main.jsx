import React from 'react';

import Nav from './components/Nav';
import Movies from './components/Movies';

class Index extends React.PureComponent {
  render() {
    return (
      <div className="d-flex">
        <div className="left-nav">
          <Nav />
        </div>
        <div className="flex-grow-1">
          <Movies />
        </div>
      </div>
    );
  }
}

export const Component = Index;
export default Index;
