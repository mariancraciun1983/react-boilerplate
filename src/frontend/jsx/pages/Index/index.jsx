import React from 'react';
import { Switch, Route } from 'react-router';

import Main from './Main';
import Genre from './Genre';

const indexRoutes = () => (
  <Switch>
    <Route component={Main} path="/" exact />
  </Switch>
);

const genreRoutes = () => (
  <Switch>
    <Route component={Genre} path="/genre/:genre_slug" />
  </Switch>
);

export default { indexRoutes, genreRoutes };
