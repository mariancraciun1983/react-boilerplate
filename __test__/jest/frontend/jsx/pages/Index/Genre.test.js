/// <reference types="jest" />
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import {Component} from '../../../../../../src/frontend/jsx/pages/Index/Genre';
import { genericWrapper } from '../../../../__mocks__/componentWrappers';

import mock from '../../../../__mocks__/store';


describe("components/pages/Index/Genre", () => {
  let props;
  beforeEach(() => {
    props =  {
      redirect: jest.fn(),
      genre: mock.fullStore.genres.list[0],
    }
  });
  it("Renders Correctly", () => {
    expect.assertions(1);
    const tree = renderer
    .create(genericWrapper(<Component {...props}/>))
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Handles bad genre", () => {
    expect.assertions(1);
    shallow(<Component {...props} genre={null}/>);
    expect(props.redirect).toHaveBeenCalledWith('/error/404');
  });

});
