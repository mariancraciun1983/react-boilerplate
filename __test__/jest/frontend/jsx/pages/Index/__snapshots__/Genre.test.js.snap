// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Index/Genre Renders Correctly 1`] = `
<div
  className="d-flex"
>
  <div
    className="left-nav"
  >
    <ul
      className="nav flex-column"
    >
      <li
        className="nav-item"
      >
        <div
          className="nav-link"
        >
          <strong>
            Genres
          </strong>
        </div>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link active"
          href="/genre/action"
          onClick={[Function]}
        >
          Action
          *
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/adventure"
          onClick={[Function]}
        >
          Adventure
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/animation"
          onClick={[Function]}
        >
          Animation
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/children"
          onClick={[Function]}
        >
          Children
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/comedy"
          onClick={[Function]}
        >
          Comedy
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/crime"
          onClick={[Function]}
        >
          Crime
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/documentary"
          onClick={[Function]}
        >
          Documentary
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/drama"
          onClick={[Function]}
        >
          Drama
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/fantasy"
          onClick={[Function]}
        >
          Fantasy
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/horror"
          onClick={[Function]}
        >
          Horror
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/imax"
          onClick={[Function]}
        >
          IMAX
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/musical"
          onClick={[Function]}
        >
          Musical
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/mystery"
          onClick={[Function]}
        >
          Mystery
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/romance"
          onClick={[Function]}
        >
          Romance
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/scifi"
          onClick={[Function]}
        >
          Sci-Fi
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/thriller"
          onClick={[Function]}
        >
          Thriller
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/war"
          onClick={[Function]}
        >
          War
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/genre/western"
          onClick={[Function]}
        >
          Western
          
        </a>
      </li>
    </ul>
  </div>
  <div
    className="flex-grow-1"
  >
    <div
      className="row"
    >
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/e00fba4e-f6d6-437c-9ad5-42b8667e4634/poseidon-adventure-the"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/1.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/e00fba4e-f6d6-437c-9ad5-42b8667e4634/poseidon-adventure-the"
              onClick={[Function]}
            >
              <strong>
                Poseidon Adventure, The
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              8.54$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/173a5d13-b294-4c26-80a8-3fbc774a1b00/better-tomorrow-iii-love-and"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/2.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/173a5d13-b294-4c26-80a8-3fbc774a1b00/better-tomorrow-iii-love-and"
              onClick={[Function]}
            >
              <strong>
                Better Tomorrow III: Love and Death in Saigon, A
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              4.07$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/2e2fb5c0-aac6-4d12-8271-b8b0fc6dc90b/9th-company"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/3.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/2e2fb5c0-aac6-4d12-8271-b8b0fc6dc90b/9th-company"
              onClick={[Function]}
            >
              <strong>
                9th Company
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              3.35$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/8e9bf7dc-1611-4303-a230-f054c5e3d986/wild-zero"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/3.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/8e9bf7dc-1611-4303-a230-f054c5e3d986/wild-zero"
              onClick={[Function]}
            >
              <strong>
                Wild Zero
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              7.46$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/79a22735-706c-486b-9da7-61ccb37ef11d/rapid-fire"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/4.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/79a22735-706c-486b-9da7-61ccb37ef11d/rapid-fire"
              onClick={[Function]}
            >
              <strong>
                Rapid Fire
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              1.63$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/b7c648f6-d224-43a8-9d47-8979c20a648c/so-undercover"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/1.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/b7c648f6-d224-43a8-9d47-8979c20a648c/so-undercover"
              onClick={[Function]}
            >
              <strong>
                So Undercover
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              8.82$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/d35cf0d8-4eba-46bd-908c-0c8fff53e1fd/thunderbirds-are-go"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/1.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/d35cf0d8-4eba-46bd-908c-0c8fff53e1fd/thunderbirds-are-go"
              onClick={[Function]}
            >
              <strong>
                Thunderbirds Are GO
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              3.76$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/42da43d8-3fd1-436c-9e8a-ac4a55f4ff48/fire-with-fire"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/2.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/42da43d8-3fd1-436c-9e8a-ac4a55f4ff48/fire-with-fire"
              onClick={[Function]}
            >
              <strong>
                Fire with Fire
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              9.95$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/604beef7-b110-4dd6-af2e-63acef910fa2/arena"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/3.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/604beef7-b110-4dd6-af2e-63acef910fa2/arena"
              onClick={[Function]}
            >
              <strong>
                Arena
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              2.82$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/ba8ddbcf-7378-44d4-8622-f0bea87e697a/fury"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/4.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/ba8ddbcf-7378-44d4-8622-f0bea87e697a/fury"
              onClick={[Function]}
            >
              <strong>
                Fury
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              9.24$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/539c4409-6ca6-417b-9ed7-76a86b71a725/american-strays"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/1.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/539c4409-6ca6-417b-9ed7-76a86b71a725/american-strays"
              onClick={[Function]}
            >
              <strong>
                American Strays
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              6.57$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3"
      >
        <a
          href="/movie/d9d20484-9a39-4e52-9039-f42265902e4a/journey-to-the-center-of-the-e"
          onClick={[Function]}
        >
          <img
            alt="Movie"
            className="img-fluid"
            src="/assets/images/movies/1.svg"
            width="100%"
          />
        </a>
        <div
          className="d-flex pt-3"
        >
          <div
            className="flex-grow-1"
          >
            <a
              href="/movie/d9d20484-9a39-4e52-9039-f42265902e4a/journey-to-the-center-of-the-e"
              onClick={[Function]}
            >
              <strong>
                Journey to the Center of the Earth
              </strong>
            </a>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              $
            </span>
            <span>
              8.36$
            </span>
            <br />
          </div>
          <div
            className="ml-auto add-cart"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
